pkg/client/factory_test.go

```diff
@@ -560,3 +560,46 @@ func TestClientWithEmptyCmabConfig(t *testing.T) {
 	assert.NotNil(t, client.DecisionService)
 	client.Close()
 }
+
+func TestCmabConfigWithCustomPredictionEndpoint(t *testing.T) {
+	// Test that custom prediction endpoint is correctly set in CmabConfig
+	customEndpoint := "https://custom.endpoint.com/predict/%s"
```

Comment:
Do we want to make sure this is a fake endpoint? Because endpoint.com is a real site.